internal/provider/resource_tfe_workspace.go

```diff
@@ -80,9 +73,12 @@ func resourceTFEWorkspace() *schema.Resource {
 			},
 
 			"agent_pool_id": {
+				Deprecated: `Please do not use this argument to set an agent pool ID.
+				Instead, configure tfe_workspace_execution_mode to attach an agent pool to a workspace.
+				This attribute may be removed in the next major version of the provider.`,
```

Comment:
This is fine but it strikes me as a little pleading instead of prescriptive.

```suggestion
				Deprecated: `Use tfe_workspace_execution_mode to configure a workspace for agent
				pool execution. This attribute may be removed in the next major version of the provider.`,
```

Reply:
Lol it's polite 🤷‍♀️